### PR TITLE
WIP [daint, dom] GROMACS patched with PLUMED and perftools

### PR DESCRIPTION
**Pull request by lucamar**
The recipe fails with the following error:
```
Failed to determine lib subdirectory in /apps/daint/UES/sandbox/lucamar/easybuild/software/GROMACS/2016.3-CrayGNU-17.08-PLUMED-2.3.0-cuda-8.0-pat-6.5.1
```
However the folder structure is built correctly: any advice?

### CODE REVIEW

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-CrayGNU-17.08-PLUMED-2.3.0-cuda-8.0-pat-6.5.1.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+description = """GROMACS is a versatile package to perform molecular dynamics,
+ i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles."""
+
+toolchain = {'name': 'CrayGNU', 'version': '17.08'}
+toolchainopts = {'usempi': 'True', 'openmp': 'True'}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
+sources = [SOURCELOWER_TAR_GZ]
+
+preconfigopts = ' pushd ../gromacs-%(version)s && plumed-patch -p --runtime --shared <<< 2 && popd && '
```
Comment (victorusu) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2016.3-CrayGNU-17.08-PLUMED-2.3.0-cuda-8.0-pat-6.5.1.eb]: @lucamar, I believe that the GROMACS' easyblock already does this operation, as soon as you add PLUMED in the dependency list.

**Comment by victorusu** [comment]
@lucamar, for the error you found related to not finding `lib`, I think you need to include the additional flags: ` -DGMX_BUILD_SHARED_EXE=off -DGMX_SIMD=AVX2_256`

**Comment by lucamar** [comment]
Thanks @victorusu , I will follow your advice and give it a try.

**Comment by lucamar** [comment]
Unfortunately the build stil fails with the same error, although it does not seem to be related to `perftools`, as it appears even if I comment out the corresponding dependency.

**Comment by lucamar** [comment]
The recipe `GROMACS-2016.4-CrayGNU-17.08-PLUMED-2.4.0-cuda-8.0.eb` works fine, but as soon as I add the following line: 
`prebuildopts = ' export CRAYPAT_COMPILER_OPTIONS=1 && export CRAYPAT_LITE=sample_profile && ' 
and thus create `GROMACS-2016.4-CrayGNU-17.08-PLUMED-2.4.0-cuda-8.0-pat-6.5.1.eb`, then I get the following error:
```
ERROR: no objects or libraries specified
FATAL: Generating link information failed
collect2: error: ld returned 1 exit status
src/gromacs/CMakeFiles/libgromacs.dir/build.make:25187: recipe for target 'lib/libgromacs.so.2.4.0' failed
make[2]: *** [lib/libgromacs.so.2.4.0] Error 1
```

**Comment by lucamar** [comment]
I'm closing this pull request, as it deals with an old release of `perftools`.

**Pull request closed without merging**